src/components/utils/__tests__/useStateMachine.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { h, FunctionComponent } from 'preact'
```

Comment:
Thanks for adding the tests in this file, they make the `useStateMachine` behaviour very clear.